### PR TITLE
Fix translated floating-base IK Jacobians

### LINKED ISSUE #3866: Floating-base free-joint analytic IK Jacobian drifts with distance from world origin (regression from #2539)
## Summary

In `newton.ik.IKJacobianType.ANALYTIC` mode, the inverse-kinematics Jacobian for a floating base (a root `JointType.FREE`, and equivalently `JointType.DISTANCE`) is wrong once the body is translated away from the world origin. The error in the analytic motion subspace grows linearly with the body's distance from the origin, so floating-base ANALYTIC IK drifts the farther the character/robot travels from the origin. In-place solves at the origin are unaffected, and `AUTODIFF` mode is correct throughout.

On a traveling motion clip we observed roughly **150 mm** of end-effector drift in ANALYTIC mode that does not appear in AUTODIFF mode.

## Affected versions

- Introduced in **1.4.0** by #2539 (commit `250fae60`).
- Still present in **1.5.0rc2** and on `main` (`1.6.0.dev0`).
- `AUTODIFF` mode is correct in all versions.

## Root cause

#2539 rewrote the FREE/DISTANCE analytic motion subspace (`write_free_distance_motion_subspace` in `newton/_src/sim/articulation.py`) so that the angular DOFs' rotation-to-translation coupling is taken about the **child body COM expressed in world coordinates**:

```python
x_child_com_world = wp.transform_point(X_wc, body_com_child)
...
joint_S_s[qd_start + 3] = wp.spatial_vector(-wp.cross(axis_world_x, x_child_com_world), axis_world_x)
```

For a floating base whose parent anchor is fixed, the origin-invariant coupling must be taken about the **fixed parent-joint anchor** (the translation of `X_pa_world`), not the traveling child COM. Because the child COM moves with the body, the `-cross(axis_world, x_child_com_world)` terms pick up a spurious contribution proportional to the body's world position, so the Jacobian's rotation-to-translation block is wrong by an amount that scales with distance from the origin. The public world-frame twist convention introduced by #2539 is otherwise correct and should be preserved.

## AUTODIFF-confirmed diagnosis

Comparing the ANALYTIC and AUTODIFF IK Jacobians for a single root FREE (or DISTANCE) body at increasing world translations shows exact agreement at the origin and a divergence that grows with translation. AUTODIFF matches finite differences at every translation; only ANALYTIC drifts. Equivalently, the analytic motion subspace `joint_S_s` for the root joint is not invariant under pure translation of the base, when it should be.

## Steps to reproduce

1. Build a model with a single root `FREE` joint (identity parent/child anchors).
2. Set the joint's translation to increasing world positions (0, 1, 5, 20 m) with an identity orientation.
3. Build two `IKSolver`s over an `IKObjectivePosition` on that body: one `AUTODIFF`, one `ANALYTIC`.
4. Compute each solver's Jacobian and compare. The max abs difference is ~0 at translation 0 and grows with translation under current `main`/1.5.0rc2.

## Minimal reproduction

```python
import numpy as np
import warp as wp
import newton
import newton.ik as ik

device = "cuda:0"

with wp.ScopedDevice(device):
    builder = newton.ModelBuilder()
    body = builder.add_link(mass=1.0)
    joint = builder.add_joint_free(
        parent=-1,
        child=body,
        parent_xform=wp.transform_identity(),
        child_xform=wp.transform_identity(),
    )
    builder.add_articulation([joint])
    model = builder.finalize(device=device, requires_grad=True)

    translations = (0.0, 1.0, 5.0, 20.0)
    joint_q = np.zeros((len(translations), model.joint_coord_count), dtype=np.float32)
    joint_q[:, 0] = translations   # translate the base along +x
    joint_q[:, 6] = 1.0            # identity quaternion (w = 1)

    targets = wp.zeros(len(translations), dtype=wp.vec3, device=device)
    objectives = [ik.IKObjectivePosition(body, wp.vec3(), targets)]

    solver_auto = ik.IKSolver(model, len(translations), objectives,
                              jacobian_mode=ik.IKJacobianType.AUTODIFF)
    solver_ana = ik.IKSolver(model, len(translations), objectives,
                             jacobian_mode=ik.IKJacobianType.ANALYTIC)

    q_auto = wp.array(joint_q, device=device, requires_grad=True)
    q_ana = wp.array(joint_q, device=device)
    solver_auto._impl._compute_residuals(q_auto)
    solver_ana._impl._compute_residuals(q_ana)
    J_auto = solver_auto._impl._jacobian_at(solver_auto._impl._ctx_solver(q_auto)).numpy()
    J_ana = solver_ana._impl._jacobian_at(solver_ana._impl._ctx_solver(q_ana)).numpy()

    for i, t in enumerate(translations):
        print(f"translation {t:>5} m: max|J_ana - J_auto| = {np.abs(J_ana[i] - J_auto[i]).max():.6e}")
```

Under current `main`/1.5.0rc2 the analytic-vs-autodiff error is ~0 at translation 0 and grows with translation (nonzero at 1/5/20 m). It should be 0 at every translation.

## Expected behavior

The ANALYTIC free/distance-joint IK Jacobian should equal the AUTODIFF Jacobian (and finite differences) regardless of the base's world position. The analytic motion subspace for a floating base must be origin-invariant: the angular DOFs' translation coupling should reference the fixed parent-joint anchor rather than the moving child COM.

### PR DESCRIPTION
**Pull request by ooctipus**
## Description

Fix distance-dependent drift in analytic IK for translated `FREE` and `DISTANCE` joints.

The public Jacobian and mass-matrix paths need motion-subspace columns referenced to the child COM, while the current LM and L-BFGS IK integrators apply generalized-coordinate updates about the parent anchor. This change makes the shared helper's pivot explicit and has each IK optimizer request the parent-anchor pivot without changing the public COM-referenced path.

Regression coverage uses nonzero parent-anchor translation and rotation, a nonzero child COM, and base translations from 0 to 20 m. It checks LM analytic Jacobians and L-BFGS analytic gradients against AUTODIFF on CPU and CUDA.

This preserves the public twist convention from #2539 and addresses the review feedback on #3868.

Closes #3866

## Checklist

- [x] New or existing tests cover these changes
- [x] The documentation is up to date with these changes
- [x] For user-facing changes, a fragment has been added by following the [changelog fragment instructions](https://github.com/newton-physics/newton/blob/main/changelog/README.md)

## Test plan

- [x] Verified the new translated-joint regressions fail before the fix (42 subtest failures).
- [x] `uv run --extra dev -m newton.tests -p "test_ik*.py" --strict-warnings` (111 passed).
- [x] `uv run --extra dev -m newton.tests -k floating_free_jacobian -k descendant_free_jacobian -k floating_free_mass_matrix -k descendant_free_mass_matrix` (12 passed).
- [x] `uvx pre-commit run -a`.
- [x] Towncrier draft build.
- [x] Isaac Lab's ANYmal-C FLAT sampler recovered `39.06%` IK acceptance (`1406 / 3600`), matching its historical baseline instead of the regressed `9.25%`.

## Bug fix

**Steps to reproduce:**

1. Build a model with a root `FREE` or `DISTANCE` joint and a nonzero parent anchor.
2. Evaluate a position objective at root translations of 0, 1, 5, and 20 m.
3. Compare the ANALYTIC derivative with AUTODIFF.
4. Observe that the difference grows with translation before this change.

**Minimal reproduction:**

Run the added `test_translated_*_jacobian_compare` cases against the parent commit. They fail for both joint types and pass with this change.

### CODE REVIEW

**Review by eric-heiden — APPROVED** [review]
Thanks, LGTM!
